api_app/analyzers_manager/file_analyzers/cuckoo_scan.py

```diff
@@ -6,39 +6,34 @@
 import requests
 import logging
 
-from api_app.exceptions import AnalyzerRunException, AnalyzerConfigurationException
+from api_app.exceptions import AnalyzerRunException
 from api_app.helpers import get_binary
 from api_app.analyzers_manager.classes import FileAnalyzer
-from intel_owl import secrets
 
 logger = logging.getLogger(__name__)
 
 
 class CuckooAnalysis(FileAnalyzer):
-    def set_config(self, additional_config_params):
+    def set_params(self, params):
         # cuckoo installation can be with or without the api_token
         # it depends on version and configuration
         self.session = requests.Session()
-        api_key_name = additional_config_params.get("api_key_name", None)
-        if api_key_name:
-            api_key = secrets.get_secret(api_key_name)
-            self.session.headers["Authorization"] = f"Bearer {api_key}"
-        else:
+        self.__api_key = self._secrets["api_key_name"]
+        if not __api_key:
             logger.info(
                 f"{self.__repr__()}, (md5: {self.md5}) -> Continuing w/o API key.."
             )
-        self.cuckoo_url = secrets.get_secret("CUCKOO_URL")
+
+        self.session.headers["Authorization"] = f"Bearer {self.__api_key}"
```

Comment:
This should be inside an `if/else` because we only want to set this header if the `api_key` is not `None`.